### ISSUE TITLE
Read ifcxml from Autodesk Revit

### ISSUE THREAD

**Issue by kms-hwi** [CAFM-Connect/Ifc.NET#5]
Hello,
we have implemented this library sucessfully in our facility management project.  We can read files coming from CAFM-Connect without any problems.
But we can't read in IFC 4 files from Revit. (We also can not read it with the CAFM-Connect Editor)

Have you got any tipps how to create a valid ifcxml file with Revit (2018)? It seems to be a problem with the xml namespaces, that revit puts into the xml.  For example:
<?xml version="1.0" encoding="utf-8"?>
<doc:iso_10303_28 xmlns:exp="urn:oid:1.0.10303.28.2.1.1" xmlns:doc="urn:oid:1.0.10303.28.2.1.3" ...
...
<IfcLengthMeasure exp:pos="0">0.</IfcLengthMeasure>

Error in CAFM-Connect Editor: "<iso_10303_28 xmlns='urn:oid:1.0.10303.28.2.1.3'> wurde nicht erwartet."

Thanks
Henrik Wilsdorf
KMS Computer GmbH

**Comment by klacol** [comment]
Good question; I am not sure, where the problem is located. In the Revit Export or the CC-Import? Could you share the whole IFCXML file?

**Comment by kms-hwi** [comment]
We first will check it with a Revit expert in the week from dec. 18th. In Revit there are many options to configure the export. 
After that I will share an sample ifcxml. (Or I can present the valid configuration for Revit IFC export)

**Comment by kms-hwi** [comment]
Here is a sample file (building with 3 rooms) from Revit 2018. We select Export - IFC - IFC4 Design Transfer View - File type IFC XML.
As you can see, they use an old XML schema IFC2x2/FINAL. So this is a basic problem in the Revit export. This is frustrating. We have to clarify it first with the Revit guys...  
Or someone has other tipps to correct this?

Edit: The FZK Ifc Viewer 4.8 can read / display this Revit xml file.

[Haus1.zip](https://github.com/CAFM-Connect/Ifc.NET/files/1571284/Haus1.zip)

**Comment by klacol** [comment]
Very nice, yes, the support for IFCXML is often not very good. The FZK Viewer can also read the file as ifczip. You can just rename the ending .zip into .ifczip and the FZK Viewer still opens it. This can be a nice move for IFCXML files, as they tend to be bigger, and zipping them reduces their size from 118 KB to 11 KB for your example. This is a reduction to 10 percent of the original size.

I use XmlSpy to make the files valid against the target schema. First, it should be clarified which version of IFC shoul be exported : Ifc 2x3 or Ifc 4?

**Comment by kms-hwi** [comment]
Yes, this is nice. It is the IFC4 schema, as we have selected at Revit IFC export.
"<doc:express id="exp_1" external="" schema_name="IFC4">"
Only the xml headers seems to be others:
This is ISO 10303-28 XML.  But you use a "clean" ifcXML with CAFM-Connect.

The Question is, if you can read this ISO 10303-28 XML in a future release?

**Comment by klacol** [comment]
Until today we implemented "only" the "clean" ifcXML with CAFM-Connect.

The standard for IfcXML provides detailed information, how to implement IfcXML with the ISO 10303-28 XML variant on this page:

http://www.buildingsmart-tech.org/specifications/ifcxml-releases/ifcxml4-release

We will think about it.

**Comment by thoffeller** [comment]
hi Klaus, have the same issue now ;-) the site is always empty ...
have tested the sample from github sample files, here's the same issue.
Cheers,
Tim
